File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
# 앞자리가 커야됨
# 앞자리가 같은 경우에는 뒷자리가 큰순, 
# 앞자리가 같은 경우에 길이가 다르다면 짧은 길이는 앞자리로 채움?

# 문자열로 바꾸기
# 가장 앞자리 순을 기준으로 정렬

# 네자리로 만들고 다
# 그걸 기존 numbers가 키값으로하고 네자리는 값으로매칭
# 
def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=lambda x: x*3, reverse=True)
    return str(int(''.join(numbers)))
# [980, 93, 9]
# 93980
# 98930]

#  네자리수로 만든다 전부

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(string a, string b){
    return a+b > b+a;
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> s;
    for (int i=0; i<numbers.size(); i++){
        s.push_back(to_string(numbers[i]));
    }
    sort(s.begin(), s.end(), cmp);
    
    for (int i=0; i<numbers.size(); i++){
        answer+=s[i];
    }
    
    if (answer[0] == '0'){
        return "0";
    }
    return answer;
}





// 1. 정수형 배열 -> 문자열 배열
//  2. 문자열 비교 함수 생성 - sort세번째인자
// 3. sort
// 4. 문자열합치기